pkg/frontend/openshiftcluster_putorpatch.go

```diff
@@ -139,7 +140,13 @@ func (f *frontend) _putOrPatchOpenShiftCluster(ctx context.Context, r *http.Requ
 		}
 
 	} else {
-		doc.OpenShiftCluster.Properties.ProvisioningState = api.ProvisioningStateUpdating
+		// TODO: Get rid of the special case
```

Comment:
I woild like to do it in a separate PR: this one is already big.

Reply:
+1, I don't see much of a neater way anyway